test/python/transpiler/test_check_map.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Test the Check Map pass"""
+
+import unittest
+
+from qiskit import QuantumRegister, QuantumCircuit
+from qiskit.transpiler.passes import CheckMap
+from qiskit.mapper import Coupling
+from qiskit.dagcircuit import DAGCircuit
+from ..common import QiskitTestCase
+
+
+class TestCheckMap(QiskitTestCase):
+    """ Tests the CheckMap pass."""
+
+    def test_trivial_map(self):
+        """ Trivial map in a circuit without entanglement
+         qr0:---[H]---
+
+         qr1:---[H]---
+
+         qr2:---[H]---
+
+         Coupling map: None
+        """
+        qr = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.h(qr)
+        coupling = Coupling()
+        dag = DAGCircuit.fromQuantumCircuit(circuit, expand_gates=False)
+        pass_ = CheckMap(coupling)
+        pass_.run(dag)
+        self.assertTrue(pass_.property_set['is_mapped'])
+
+    def test_true_map(self):
+        """ Mapped is easy to check
+         qr0:--(+)-[H]-(+)-
+                |       |
+         qr1:---.-------|--
+                        |
+         qr2:-----------.--
+
+         Coupling map: [1]--[0]--[2]
+        """
+        qr = QuantumRegister(3, 'qr')
+        circuit = QuantumCircuit(qr)
+        circuit.cx(qr[0], qr[1])
+        circuit.h(qr[0])
+        circuit.cx(qr[0], qr[2])
+        coupling = Coupling({0: [1, 2]})
```

Comment:
After #1141, we should remember to update all constructors of Coupling to take list, as the dict will be deprecated in that PR.

Reply:
#1141 still extends `Coupling` but stills supports `couplingdict=`.  I made it explicit in 677db53